embedded-graphics/src/draw_target.rs

```diff
@@ -0,0 +1,304 @@
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord.x + coord.y * 64;
+///         self.framebuffer[index as usize] = color.luma();
+///
+///         Ok(())
+///     }
+///
+///     fn size(&self) -> Size {
+///         Size::new(64, 64)
+///     }
+///
+///     /// Use the accelerated method when drawing rectangles
+///     fn draw_rectangle(
+///         &mut self,
+///         item: &Styled<Rectangle, PrimitiveStyle<Gray8>>,
+///     ) -> Result<(), Self::Error> {
+///         self.fast_rectangle(item);
+///
+///         Ok(())
+///     }
+/// }
+///
+/// let mut display = FastExampleDisplay {
+///     framebuffer: [0; 4096],
+///     iface: SPI1,
+/// };
+///
+/// // Draw a rectangle from (10, 20) to (30, 40) with a white stroke
+/// let rect = egrectangle!(
+///     top_left = (10, 20),
+///     bottom_right = (30, 40),
+///     style = primitive_style!(stroke_color = Gray8::WHITE)
+/// );
+/// rect.draw(&mut display)?; // Uses the accelerated draw_rectangle function
+///
+/// // Update the display
+/// display.flush().expect("Failed to send data to display");
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// [`Drawable`]: ../drawable/trait.Drawable.html
+/// [`Pixel`]: ../drawable/struct.Pixel.html
+/// [`draw_pixel`]: ./trait.DrawTarget.html#method.draw_pixel
+/// [`DrawTarget`]: ./trait.DrawTarget.html
+pub trait DrawTarget<C>
+where
+    C: PixelColor,
+{
+    /// Error type to return when a draw or display change operation fails.
+    ///
+    /// This error is returned when a drawing operation fails, or when an error updating/clearing
+    /// the display occurs. The examples in this crate use `core::convert::Infallible`, however a
+    /// more descriptive error type should be used with real hardware.
+    type Error;
+
+    /// Draws a pixel on the display.
+    ///
+    /// Note that some displays require a "flush" operation to actually write changes to the
+    /// framebuffer.
+    fn draw_pixel(&mut self, item: drawable::Pixel<C>) -> Result<(), Self::Error>;
+
+    /// Draws an object from an iterator over its pixels.
+    fn draw_iter<T>(&mut self, item: T) -> Result<(), Self::Error>
+    where
+        T: IntoIterator<Item = drawable::Pixel<C>>,
+    {
+        for pixel in item {
+            self.draw_pixel(pixel)?;
+        }
+
+        Ok(())
+    }
+
+    /// Returns the dimensions of the `DrawTarget` in pixels.
+    fn size(&self) -> Size;
+
+    /// Clears the display with the supplied color.
+    ///
+    /// This default implementation should be replaced if the implementing driver provides an
+    /// accelerated clearing method.
+    fn clear(&mut self, color: C) -> Result<(), Self::Error>
+    where
+        Self: Sized,
+    {
+        primitives::Rectangle::new(Point::zero(), Point::zero() + self.size())
+            .into_styled(PrimitiveStyle::with_fill(color))
+            .draw(self)
+    }
+
+    /// Draws a line primitive.
```

Comment:
As mentioned above, in the "Hardware Acceleration" example, we should have a consistent style of referring to styled vs. raw primitives.
This also applies to the other `draw_...` methods.